### PR TITLE
migration: retry on celery ResponseError

### PR DESCRIPTION
**Pull request by pazembrz**
*INSPIR-3618

### CODE REVIEW

File: backend/inspirehep/migrator/tasks.py [hunk 1]
```diff
@@ -320,8 +331,8 @@ def create_records_from_mirror_recids(self, recids):
     ignore_results=False,
     queue="migrator",
     acks_late=True,
-    retry_backoff=2,
-    retry_kwargs={"max_retries": 6},
+    retry_backoff=RETRY_BACKOFF,
+    retry_kwargs={"max_retries": MAX_RETRY_COUNT},
     autoretry_for=(OperationalError,),
```
Comment (michamos) [inline backend/inspirehep/migrator/tasks.py:336]: why not simply add it to the `autoretry`?

File: backend/inspirehep/migrator/tasks.py [hunk 2]
```diff
@@ -504,11 +515,30 @@ def migrate_record_from_mirror(
...
+    LOGGER.info(
+        f"Waiting for task completion.", waiting_task_id=task, retry_count=retry_count
+    )
+    try:
+        next_task = AsyncResult(task).get()
+    except ResponseError:
+        if retry_count < MAX_RETRY_COUNT:
+            LOGGER.exception(
+                "Redis ResponseError Exception, retrying",
+                task=task,
```
Comment (michamos) [inline backend/inspirehep/migrator/tasks.py:530]:
```suggestion
                waiting_task_id=task,
```
to be consistent with the rest of logging (`task` is the name of the task)

File: backend/inspirehep/migrator/tasks.py [hunk 3]
```diff
@@ -504,11 +515,30 @@ def migrate_record_from_mirror(
...
-    LOGGER.info(f"Waiting for task completion.", waiting_task_id=task)
-    next_task = AsyncResult(task).get()
+    LOGGER.info(
+        f"Waiting for task completion.", waiting_task_id=task, retry_count=retry_count
+    )
+    try:
+        next_task = AsyncResult(task).get()
+    except ResponseError:
+        if retry_count < MAX_RETRY_COUNT:
+            LOGGER.exception(
```
Comment (michamos) [inline backend/inspirehep/migrator/tasks.py:528]: do we want to log this as an exception too?

File: backend/inspirehep/migrator/tasks.py [hunk 4]
```diff
@@ -504,11 +515,30 @@ def migrate_record_from_mirror(
...
+                "Redis ResponseError Exception, retrying",
+                task=task,
+                retry_count=retry_count,
+            )
+            time.sleep(RETRY_BACKOFF * retry_count + 1)
+            return wait_for_all_tasks(task, retry_count + 1)
+        else:
+            LOGGER.exception(
+                "Redis ResponseError Exception failed too many times times.",
+                task=task,
```
Comment (michamos) [inline backend/inspirehep/migrator/tasks.py:538]: same

Comment (pazembrz) [reply backend/inspirehep/migrator/tasks.py:538]: Yes, this one we want to log as exception as it's failing here and stops after too many retries.

File: backend/inspirehep/migrator/tasks.py (hunk 3 above)
Comment (pazembrz) [reply backend/inspirehep/migrator/tasks.py:528]: :+1:  Yes, you are right. Here info would be enough.

File: backend/inspirehep/migrator/tasks.py (hunk 2 above)
Comment (pazembrz) [reply backend/inspirehep/migrator/tasks.py:530]: :+1:

File: backend/inspirehep/migrator/tasks.py (hunk 4 above)
Comment (michamos) [reply backend/inspirehep/migrator/tasks.py:538]: I meant `task` -> `waiting_task_id` here

Comment (pazembrz) [reply backend/inspirehep/migrator/tasks.py:536]: :+1:

File: backend/inspirehep/migrator/tasks.py (hunk 1 above)
Comment (pazembrz) [reply backend/inspirehep/migrator/tasks.py:336]: As discussed IRL Micha misread code.

**Pull request merged**